lib/charms/opensearch/v0/helper_networking.py

```diff
@@ -54,11 +54,12 @@ def units_ips(charm: CharmBase, peer_relation_name: str) -> Dict[str, str]:
     return unit_ip_map
 
 
-def is_reachable(ip_address: str, port: int) -> bool:
+def is_reachable(host: str, port: int) -> bool:
     """Attempting a socket connection to a host/port."""
     s = socket.socket()
+    s.settimeout(5)
     try:
-        s.connect((ip_address, port))
+        s.connect((host, port))
         return True
     except Exception:
```

Comment:
Agreed. 
However, The socket connection can throw so many types of exceptions, that are either subclasses of `OSError` or of `Exception` i.e: `socket.error`.
I thought in this specific case, it makes sense to just go generic, since they all mean the same thing in the context of this function, which is the fact the node is not reachable.

Reply:
hmmm, i think it would be better to do: `Except (OSError,socket.error) as e` but also I trust you here!